libs/test/SalesforceReactTest/src/com/salesforce/androidsdk/reactnative/util/ReactTestActivity.java

```diff
@@ -39,23 +40,23 @@
  */
 public class ReactTestActivity extends SalesforceReactActivity {
 
-    static String username = "__USER_NAME__";
-    static String accountName = "__ACCOUNT_NAME__";
-    static String refreshToken = "__REFRESH_TOKEN__";
+    static String username = TestCredentials.USERNAME;
+    static String accountName = TestCredentials.ACCOUNT_NAME;
+    static String refreshToken = TestCredentials.REFRESH_TOKEN;
     static String authToken = "--will-be-set-through-refresh--";
-    static String identityUrl = "__IDENTITY_URL__";
-    static String instanceUrl = "__INSTANCE_URL__";
-    static String loginUrl = "__LOGIN_URL__";
-    static String orgId = "__ORG_ID__";
-    static String userId = "__USER_ID__";
-    static String photoUrl = "__PHOTO_URL__";
+    static String identityUrl = TestCredentials.IDENTITY_URL;
+    static String instanceUrl = TestCredentials.INSTANCE_URL;
+    static String loginUrl = TestCredentials.LOGIN_URL;
+    static String orgId = TestCredentials.ORG_ID;
+    static String userId = TestCredentials.USER_ID;
+    static String photoUrl = TestCredentials.PHOTO_URL;
+    static String clientId = TestCredentials.CLIENT_ID;
 
     @Override
     protected ClientManager buildClientManager() {
         final ClientManager clientManager = super.buildClientManager();
-        final ClientManager.LoginOptions loginOptions = SalesforceSDKManager.getInstance().getLoginOptions();
         clientManager.createNewAccount(accountName, username, refreshToken, authToken, instanceUrl,
-                loginUrl, identityUrl, loginOptions.getOauthClientId(), orgId, userId,
+                loginUrl, identityUrl, clientId, orgId, userId,
```

Comment:
That's important - loginOptions.getOauthClientId() returns the client id from the bootconfig / which might not match the actual client id in test_credentials.json => not only refresh will fail, but the refresh token will be invalidated!

Reply:
So this is what was breaking our tests all this while?